lib/queue.js

```diff
@@ -63,16 +63,19 @@ var Queue = function Queue(name, redisPort, redisHost, redisOptions){
     redisOptions.db = redisOpts.DB;
   }
 
-  if (_.isString(redisPort)) {
+  if (!_.isObject(redisPort) && _.isString(redisPort)) {
```

Comment:
Ah I completely misunderstood. Yeah that makes sense since the first block might change redisPort.